Controllers/FridgeController.cs

```diff
@@ -0,0 +1,160 @@
+﻿using AutoMapper;
+using Contracts;
+using Dapper;
+using Entities.DataTransferObjects;
+using Entities.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Data;
+using System.Linq;
+
+namespace FridgeProducts.Controllers
+{
+    [Route("api/fridge")]
+    [ApiController]
+    public class FridgeController : ControllerBase
+    {
+        private readonly IRepositoryManager _repository;
+        private readonly ILoggerManager _logger;
+        private readonly IMapper _mapper;
+        private readonly DapperContext _context;
+
+        public FridgeController(IRepositoryManager repository, ILoggerManager logger, IMapper mapper, DapperContext context)
+        {
+            _repository = repository;
+            _logger = logger;
+            _mapper = mapper;
+            _context = context;
+        }
+
+        [HttpGet(Name = "GetFridges"), Authorize(Roles = "User, Administrator")]
+        public IActionResult GetFridges()
+        {
+                var fridges = _repository.Fridge.GetAllFridges(trackChanges: false);
+
+                var fridgesDTO = _mapper.Map<IEnumerable<FridgeDTO>>(fridges);
+
+                var models = _repository.Model.GetAllFridgeModels(trackChanges: false);
+
+                var fridgeswithmodels = _repository.Fridge.JoinFridgeWithModel(fridgesDTO, models);
+
+                return Ok(fridgeswithmodels);
+        }
+
+        [HttpGet("productswithzeroquantity"), Authorize(Roles = "Administrator")]
+        public async Task<IEnumerable<FridgeProductsDTO>> SetQuantity()
+        {
+            var procedurename = "FindProductsWithZeroQuantity";
+
+                var connection = _context.CreateConnection();
+               
+                var products = await connection.QueryAsync<fridge_products>(procedurename, commandType: CommandType.StoredProcedure);
+
+                var productsDTO = _mapper.Map<IEnumerable<FridgeProductsDTO>>(products);
+                
+                return productsDTO.ToList();
+        }
+
+        [HttpPost(Name = "AddFridge"), Authorize(Roles = "Administrator")]
+        public IActionResult AddFridge([FromBody] FridgeForAddingDTO Fridge)
+        {
+            if (!ModelState.IsValid)
+            {
+                _logger.LogError("Invalid model state for the FridgeForAddingDTO object");
+                return UnprocessableEntity(ModelState);
+            }
+
+            if (Fridge == null)
+            {
+                _logger.LogError("FridgeDTO object sent from client is null.");
+                return BadRequest("FridgeDTO object is null");
+            }
+
+            var FridgeEntity = _mapper.Map<fridge>(Fridge);
+
+            var modelEntity = _mapper.Map<fridge_model>(Fridge);
+
+            var CheckFridgeforRepeatByName = _repository.Fridge.FindDublicateFridgeByName(Fridge, trackChanges: false);

```

Comment:
You can just find the first fridge with the same name, if it doesn't exist than you can add fridge
And don't use FindDublicateFridge